File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let diff = 0;
    let map = {};
    let arr = [];
    for(let i =0; i<nums.length; i++){
        diff = target - nums[i];
        if(map.hasOwnProperty(diff)){
         return [map[diff],i]

        }
        else{
            map[nums[i]] = i;
            };
    }
    return [];
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for i in range(len(nums)):
            diff =target - nums[i]
            if diff in map:
                return [map.get(diff), i]
            else:
                map[nums[i]] = i